### PR TITLE
Refresh course setting page on error

### PR DESCRIPTION
**Pull request by xinyis991105**
## Description
This PR allows a course setting page to refresh automatically upon receiving erroneous setting information.

## Motivation and Context
#1385 

## How Has This Been Tested?

See the bug report. After following the steps to reproduce, you should now observe the expected behavior.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://github.com/autolab/docs)
- [ ] I have updated the documentation accordingly, included in this PR

## Other issues / help required

If unsure, feel free to submit first and we'll help you along.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -54,7 +54,7 @@ RUN mkdir -p /home/app/webapp/log && \
 USER app
 
 # precompile the Rails assets
-RUN RAILS_ENV=production bundle exec rails assets:precompile
+RUN RAILS_ENV=development bundle exec rails assets:precompile
```
Comment (fanpu) [inline Dockerfile:57]: Could I check why this change is necessary? It will impact the docker-compose installation

**Review by fanpu — CHANGES_REQUESTED** [review]
Other than reverting this line, everything else LGTM! Reproduced test plan

File: Dockerfile (hunk 1 above)
Comment (xinyis991105) [reply Dockerfile:57]: Sorry about that! I was testing my local docker compose! I must've accidentally added this commit. Will remove right away.

**Review by fanpu — APPROVED** [review]
